### PR TITLE
Honor frame number limit in GDB core unwinder

### PR DESCRIPTION
**Pull request by mhabrnal**
Related to #1260074

### CODE REVIEW

File: tests/core_stacktrace.at [hunk 1]
```diff
@@ -197,3 +197,167 @@ main(void)
...
+  char *error_message = NULL;
+
+  char *coredump_path = generate_coredump(257);
+
+  struct sr_strbuf *gdb_output = get_backtrace(coredump_path, argv[0]);
+  assert(gdb_output);
+  assert(gdb_output->buf);
+
+  char *error;
+  sr_string_to_file("/tmp/gdb_backtrace", gdb_output->buf, &error);
```
Comment (jfilak) [inline tests/core_stacktrace.at]: Please, use an unique temporary file.

**Comment by mhabrnal** [comment]
retest this please

**Comment by mhabrnal** [comment]
retest this please

**Comment by mhabrnal** [comment]
retest this please

**Pull request merged**